pulseapi/profiles/views.py

```diff
@@ -52,15 +60,47 @@ def put(self, request, *args, **kwargs):
         much mutually exclusive patterns. A try/pass make far more sense.
         '''
 
+        payload = request.data
+
         try:
-            thumbnail = request.data['thumbnail']
+            thumbnail = payload['thumbnail']
             # do we actually need to repack as ContentFile?
             if thumbnail['name'] and thumbnail['base64']:
                 name = thumbnail['name']
                 encdata = thumbnail['base64']
                 proxy = ContentFile(base64.b64decode(encdata), name=name)
-                request.data['thumbnail'] = proxy
+                payload['thumbnail'] = proxy
+        except:
+            pass
+
+        '''
+        Is there an 'enable_extended_information' property? If so,
+        this is something that may only be toggled by admins and
+        cannot be toggled through general profile updates
+        '''
+
+        removeKey(payload, 'enable_extended_information')
+
+        '''
+        Do a prefetch on the profile to update. Is the
+        'enable_extended_information' property set to False?
+        In that case, remove any extended properties from the
+        request payload.
+        '''
+
+        try:
```

Comment:
this looks like logic that should probably be in the serializer instead. So you can either tack this on to the `init` method (might be complicated) or instead just override the serializer's `to_internal_representation` method.

Reply:
Not sure I agree - this is check is based on a user associated with this specific API call, making the entry point for that call the right place to perform it. The serializer on the other hand should not know about "why" it's being called; it might be due to an HTTP request but it might just as easily be part of a code path that has nothing to do with HTTP interactions, so there is no reason to assume that a serializer has a request object to work with.

Reply:
hmm, I don't follow...the serializer would still not know why it is being called because it wouldn't deal with a user at all. All the serializer would care about is its own data. I'm not suggesting we copy/paste this logic as is to the serializer...we would really only need lines 95-99. So the method would look something like this:
```python
def to_internal_value(self, data):
    enable_extended_information = self.fields['enable_extended_information'].get_value(data)
    profile = super(UserProfileSerializer, self).to_internal_value(data)
    if enable_extended_information is not True:
        del profile['user_bio_long']
        del profile['program_type']
        ....
    return profile
```

Reply:
ah, I see.